index.js

```diff
@@ -186,10 +186,11 @@ function fromArrayBuffer (that, array) {
   if (Buffer.TYPED_ARRAY_SUPPORT) {
     // Return an augmented `Uint8Array` instance, for best performance
     array.byteLength
-    that = Buffer._augment(new Uint8Array(array))
+    that = new Uint8Array(array)
+    that.__proto__ = Buffer.prototype
   } else {
     // Fallback: Return an object instance of the Buffer class
-    that = fromTypedArray(that, new Uint8Array(array))
+    that = fromTypedArray(that, array)
```

Comment:
This will break if the user passes an ArrayBuffer into the Buffer object implementation. `fromTypedArray` takes an array-like object.